src/components/App/App.jsx

```diff
@@ -40,46 +137,182 @@ export default function App() {
+      })
+      .catch(handleMainError);
+  };
+
+  const handleSearchSubmit = (searchInput) => {
+    setSearchPreloaderVisible(true);
+    setIsSearchNotFoundVisible(false);
+    setCards([]);
+    setCardsToShow([]);
+    localStorage.removeItem('cards');
+    newsApi
+      .everything(searchInput)
+      .then(({ articles, status }) => {
+        if (status !== 'ok') {
+          throw new Error('NewsApi Error');
+        }
+        if (articles.length === 0) {
+          setSearchPreloaderVisible(false);
+          setIsSearchNotFoundVisible(true);
+          return;
+        }
+        const cardListData = [];
+        articles.forEach((element) => {
+          cardListData.push({
+            keyword: searchInput,
+            title: element.title,
+            text: element.description,
+            date: setCardDateFormat(element.publishedAt),
+            source: element.source?.name,
+            link: element.url,
+            image: element.urlToImage || imageNotAvailable,
+          });
+        });
+        setCards(cardListData);
+        localStorage.setItem('cards', JSON.stringify(cardListData));
+        setSearchPreloaderVisible(false);
+      })
+      .catch(handleMainError);
+  };
+
+  const handleShowMoreCards = () => {
+    showMoreCards();
+  };
+
+  const handleCardRemoveClick = (card) => {
+    if (card) {
+      mainApi
+        .deleteArticle(card._id)
+        .then((res) => {
+          setSavedCards((state) => state.filter((currentCard) => currentCard._id !== res._id));
+        })
+        .catch(handleMainError);
+    }
+  };
+
   return (
-    <div className="app">
-      <Header
-        loggedIn={false}
-        OnSignInClick={handleSignInClick}
-        OnSignUpClick={handleSignUpClick}
-        hideMobileMenuButton={hideMobileMenuButton}
-      />
-      <Routes>
-        <Route path="/" element={<Home />} />
-        <Route path="/SavedArticles" element={<SavedArticles />} />
-        <Route path="*" element={<NotFound />} />
-      </Routes>
-      <Footer />
-      <SignInPopup
-        onClose={closeAllPopups}
-        title="Sign In"
-        isOpen={isSignInPopupOpen}
-        /*   onSubmit={handleSignInSubmit} */
-        onSignUpPopupClick={handleSignUpClick}
-      />
-      <SignUpPopup
-        onClose={closeAllPopups}
-        /*   onSubmit={handleSignUpSubmit} */
-        title="Sign Up"
-        isOpen={isSignUpPopupOpen}
-        onSignInPopupClick={handleSignInClick}
-      />
-      <PopupWithMessage
-        title="Registration successfully completed!"
-        onClose={closeAllPopups}
-        isOpen={isPopupWithMessageOpen}
-      >
-        <Link
-          onClick={handleSignInClick}
-          className="popup__link popup__link_type_message"
+    <CurrentUserContext.Provider value={currentUser}>
+      <div className="app">
+        <Header
+          OnSignOutClick={handleSignOutClick}
+          OnSignInClick={handleSignInClick}
+          OnSignUpClick={handleSignUpClick}
```

Comment:
```js

          OnSignOutClick=
          OnSignInClick=
          OnSignUpClick=

```

these attributes should be in **lower case**

Every variable in the project should be in `camelCase` ( in lower case if it's not a class or component)